posthog/models/event.py

```diff
@@ -146,10 +147,18 @@ def filter_by_element(self, filters: Dict, team_id: int):
             filter = {}
 
         for key in ["tag_name", "text", "href"]:
-            if filters.get(key):
-                filter["element__{}".format(key)] = filters[key]
-
-        if not filter:
+            vals = filters.get(key)
```

Comment:
I think we can simplify this code. The `did_filter`, etc logic is kind of awkward.

WDYT about this:

```python
filters = Q()
for key in ["tag_name", "text", "href"]:
    values = filters.get(key, [])
    values = values if isinstance(values, list) else [values]
    if len(values) == 0: continue

    condition = Q()
    for searched_value in values:
        condition |= Q(**{"element__{}".format(key): searched_value}
    filters &= condition

if not filters: return {}
groups = groups.filter(filters)
```

It does have some slight difference of behavior if filters[key] is None - I think that's not possible?

Reply:
Ah this `None` issue was the problem! I tried something similar above last night but it was breaking a lot of tests so i left the smelly code in. The action steps that are passed in have None for ["tag_name", "text", "href"] so it does need to be handled. Thanks for the pointer